main-app/app/scripts/providers/generate-deck.js

```diff
@@ -0,0 +1,24 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.SnapCardGame.Deck')
+        .factory('GenerateDeck', ['ShuffleAndDeal', function (ShuffleAndDeal) {
```

Comment:
1. This should really just be named "DeckFactory"
2. When you have split the shuffle & deal services out (note later) inject the *shuffle* here, not the controller
3. The factory should return a deck object - with a created deck that is capable of being shuffled using the shuffle service. The Controller should be able to do something like:
    var deck = DeckFactory.Create();
    deck.shuffle();
Then deal....